src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+    throw new IllegalArgumentException("numberOfTorpedos");
     }
 
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -=- numberOfTorpedos;
```

Comment:
You probably wanted to write `-= n`, but you wrote `-= -n` instead.

Reply:
Oh, SonarLint was finally stopped complaining after this change, so I thought it was ok now. But agreed, let's write `-=`, that looks simpler :D.